src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+public class MovieRecommender {
+    private String file;
+    private Hashtable<String, Integer> HashProduct = new Hashtable<String, Integer>();
+    private Hashtable<Integer, String> InvertedHashProduct = new Hashtable<Integer, String>();
+    private Hashtable<String, Integer> HashUser = new Hashtable<String, Integer>();
+    private int users =0, products =0, reviews = 0;
+
+    public MovieRecommender(String file) throws IOException{
+        this.file = file;
+        getData();
+    }
+
+    public String getData() throws IOException {
+        int thisProduct =0, thisUser =0;
+        Files.deleteIfExists(Paths.get("Result.csv"));
+        File result = new File("Result.csv");
+        InputStream fileReader = new GZIPInputStream(new FileInputStream(this.file));
+        BufferedReader br = new BufferedReader(new InputStreamReader(fileReader));
+        FileWriter fileWriter = new FileWriter(result);
+        BufferedWriter bw = new BufferedWriter(fileWriter);
+        String line;
+        String[] sp;
+        String key, value;
+
+        while((line = br.readLine()) != null) {
+            if (line.length() >= 0) {
+                sp = line.split(" ");
+                key = sp[0];
+                if (key.equals("product/productId:")) {
+                    value = sp[1];
+                    if (!HashProduct.containsKey(value)){
+                        HashProduct.put(value,products);
+                        InvertedHashProduct.put(products,value);
+                        thisProduct = HashProduct.get(value);
+                        products++;
+                    }else{
+                        thisProduct = HashProduct.get(value);
+                    }
+                }else if (key.equals("review/userId:")){
+                    value = sp[1];
+                    if (!HashUser.containsKey(value)){
+                        HashUser.put(value, users);
+                        thisUser = HashUser.get(value);
+                        users++;
+                    }else{
+                        thisUser = HashUser.get(value);
+                    }
+                }else if (key.equals("review/score:")){
+                    String score = sp[1];
+                    bw.write(thisUser + "," + thisProduct + "," + score + "\n");
+                    reviews ++;
+                }
+            }
+        }
+        br.close();
```

Comment:
I don't like this is not in a finally block. have you heard about try-catch-with-resources? please investigate about it.